arch/arm/boot/dts/bcm283x.dtsi

```diff
@@ -327,7 +327,7 @@
 			status = "disabled";
 		};
 
-		i2c0: i2c@7e205000 {
+		i2c10: i2c@7e205000 {
```

Comment:
I'm not keen on the name `i2c10` - it doesn't convey much information - but I'm guessing you want to change it because the firmware expects the `i2c0` label to identify the header-accessible I2C0 pins (you don't say). `i2c0hw`, `i2c0if` or even `_i2c0` would be more obvious. But you should consider changing the firmware, since you will no longer be using `i2c_vc` from overlays.

Reply:
And on the subject of the firmware and it's arm<->vc remapping, what is your proposal for the rev1 model B (&A?) boards that use I2C1 for the camera? Are we orphaning them?

Reply:
Technically it's bsc0.
Yes, i2c0 needs to point at the GPIOs 0&1.

It's the weirdness that this base node gets registered as an /dev/i2cN node, but it wants to be up out the way and not used because it will just adopt the pinctrl of the last muxed access. i2c10 puts it out the way. AFAIK you can't create it as /dev/i2c0hw - it has to be a pure numeric suffix.

Hopefully ```dtparam=i2c_vc``` from the firmware should still enable the complete infrastructure of bsc0 and the mux (I think I updated both status values).
Use of ```i2c_vc``` from an overlay should be equivalent to selecting i2c0 on GPIOs 0&1 (ie the HAT ID pins).

I did make a comment about needing to refine rev1 model B's. There I think we have to remove the mux node totally and revert back to the old scheme. I believe we can do that via a ```/delete-node/```, as is done for pixelvalves on 2711 https://github.com/raspberrypi/linux/blob/rpi-5.4.y/arch/arm/boot/dts/bcm2711-rpi-4-b.dts#L188, and i2c_vc is inherently the CSI port as there were no HAT ID pins.
I'm not sure if we can get the aliases to all point to the right places.

Reply:
Minor addendum to the above as I hit "comment" too soon.
Currently the Pi4 HDMI I2C interfaces are not fixed, so become i2c11 and i2c12. I don't know if we want to lock those down, and if so to what numbers.

Reply:
So the motivation is to arrange that the /dev device name is still i2c0, while the kernel is trying to match the DT names to the device numbers? That makes sense - I'll have a ponder.

Reply:
Yes, I wanted to leave any existing users with the same behaviour and functionality as they currently get, hopefully without them having to make any changes.

The exception will be for those who used the i2c0-bcm2708 overlay (or similar) to deliberately remap i2c0 to 28&29 or 44&45 for some other reason. That's not possible whilst also exposing i2c on both sets of gpios.

The camera/display I2C is currently going to end up as i2c9 due to the DT alias.

So it's what the additional (redundant) parent i2c device gets called that is the issue.

The i2c core does look through all devices to work out the highest statically allocated bus number, and all dynamically assigned busses start from that. We could leave this parent node as a dynamically assigned bus number.

Reply:
[ Back on this after an adventure in CMA-related non-booting ]

> Currently the Pi4 HDMI I2C interfaces are not fixed, so become i2c11 and i2c12. I don't know if we want to lock those down, and if so to what numbers.

Do we want users accessing the HDMI I2C interface other than via the HDMI driver?

Reply:
We had the ```i2c2_iknowwhatimdoing``` dtparam for Pi0-3. There's no reason not to have an equivalent for Pi4.
Some people want to do DDC directly over I2C to their monitors as some allow control of brightness/contrast type parameters over.